### ISSUE TITLE
Clients cannot walk on tiles where explosions took place

### ISSUE THREAD

**Issue by fomalsd** [unitystation/unitystation#894]
# Preliminary information

## Category
BUG

## Bug: Release or Development?
0.2.2 beta 4 steam

# Report
## Current Behavior
You cannot step on tiles (you lerp back) explosives were located on (prior to their explosion)

## Expected/Wanted/Requested Behavior
Want to walk free

## Steps to Reproduce
Please enter the steps to reproduce the bug or behaviour:
1. Explode a fuel tank from safe distance
2. Try to step on the tile where the tank was
![2018-01-20 15_48_31](https://user-images.githubusercontent.com/10403536/35183646-9d64e28e-fdfa-11e7-9f9d-2320eca07835.gif)

## OS used:
OSX 10.9

## System specifications
MBP mid2010

## Network
beta server 60ms

**Comment by krille90** [comment]
can't reproduce on local host & client. Might be an issue with the headless server.

**Comment by DooblyNoobly** [comment]
happens on beta 5. All that needs to happen is the object to unregister itself from matrix both on the client and on the server (I'm assuming problem is serverside and it is not being unregistered there, that is why when you host you don't see the problem because the host is a client)

**Comment by fomalsd** [comment]
sadly still a thing on true headless

**Comment by chairbender** [comment]
The server even has an issue with exploding tanks...you can actually walk over and push the invisible tank after the explosion disappears and it suddenly becomes visible, then vanishes after a few seconds.

I'll try to fix the various issues surrounding the cleanup of exploded tanks including this one.

Mostly just seems to be due to not updating the RegisterTile so that it is marked as Passable.

**Comment by fomalsd** [comment]
yeah, I think that's just the matter of proper destroying/unregistering the fuel tank

**Comment by chairbender** [comment]
there's actually some sort of race condition, the error is non-deterministic. But I think I'm on the path to the solution finally.

**Comment by chairbender** [comment]
As part of the health refactor #1397 , fuel tank will no longer explode on damage. 

Therefore, I will focus this ticket on fixing the underlying issues that are causing this problem, as they exist outside of the fuel tank health code. I've fixed how it works for client / server but I still get the issue on headless.

Any changes I made to the fuel tank health behavior / explodewhenshot can be discarded when the health refactor comes in, but I'll leave them in for now just in case.
